### PR TITLE
Use more proper DATA_DIR in conftest.py

### PR DESCRIPTION
**Pull request by yd8534976**
In `init_env()`, `DATA_DIR` has been set to `blueoil_dir`. It is not proper. Since all dataset loaders except those who inherits `StoragePathCustomizable` will be broken. (They don't deal with `train_path` and `test_path` in config file.)
This PR makes an ordinary dataset loader can be used in `e2e` tests.
https://github.com/blue-oil/blueoil/blob/b45d0806a7daf8bff4af10e258c1e0b9d008953b/tests/e2e/conftest.py#L27-L33

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: tests/e2e/conftest.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def init_env():
 
     # TODO: Remove this setting after lmnet.environment has been refactored.
     envs = {
-        "DATA_DIR": blueoil_dir,
+        "DATA_DIR": os.path.join(blueoil_dir, "lmnet", "tests", "fixtures", "datasets"),
```
Comment (iizukak) [inline tests/e2e/conftest.py]:
```suggestion
        "DATA_DIR": os.path.join(blueoil_dir, "lmnet/tests/fixtures/datasets"),
```

Comment (tsawada) [reply tests/e2e/conftest.py]: I think we should do `os.path.join("a", "b", "c")` instead of `("a", "b/c")` for better cross-platform compatibility

Comment (iizukak) [reply tests/e2e/conftest.py]: I see.

**Pull request merged**